### PR TITLE
Align and fix some Windows MSVC build setting

### PR DESCRIPTION
**Pull request by alef**
#### Summary
Build "Align and fix some Windows MSVC build setting"

#### Purpose of change
Mostly align CMake linker settings, see below.

#### Describe the solution
* Drop that stupid hack of mine to force Ninja Build on Windows Terminal
* Keep `msvc-full-features` Solution file flags aligned into CMake's
* Merge Windows manifests for `utf-8` and UAC settings
* Avoid CMake's MVSC build failing for duplicated resource ID
* Edit: rework MSYS2/MINGW64 static linking in the process

#### Testing
Built with Solution file and CMake presets. The icon is there, the `.exe` contains the manifest. It runs. *NOT tested* the msys2/mingw build.

#### Additional context
I am not sure the UAC section and UTF-8 thing can coexist together this way.
I hope @Qrox and @akrieger review this.

### CODE REVIEW

File: src/CMakeLists.txt [hunk 1]
```diff
@@ -51,6 +51,10 @@ if (TILES)
                 ${MAIN_CPP}
                 ${MESSAGES_CPP}
                 ${RESOURCE_RC})
+        set_property(TARGET cataclysm-tiles PROPERTY
+        LINK_OPTIONS
+            /MANIFESTINPUT:${CMAKE_SOURCE_DIR}/data/application_manifest.xml
+        )
```
Comment (Qrox) [inline src/CMakeLists.txt:57]: This probably won't work for MinGW.

Comment (alef) [reply src/CMakeLists.txt:57]: The `Makefile` build, as by docs, works.
I will attempt `CMake` without profiles to build under `MSYS2` with `MSYSTEM=mingw64`.

File: src/resource.rc [hunk 2]
```diff
@@ -3,6 +3,8 @@
 
 #include <windows.h>
 0  ICON  "../data/cataicon.ico"
+#ifndef CMAKE
```
Comment (Qrox) [inline src/resource.rc:6]: Need to change the condition here to match the one in `src/CMakeLists.txt`.

File: build-scripts/show-manifest [hunk 3]
```diff
@@ -0,0 +1,3 @@
+rm -f manifest.xml
+"C:/Program Files (x86)/Windows Kits/10/bin/10.0.19041.0/x86/mt.exe" "-inputresource:$1;#1" "-out:manifest.xml"
```
Comment (akrieger) [inline build-scripts/show-manifest:2]: This path is not portable/stable, different versions of the windows 10 kit will have different paths to this.

File: build-scripts/show-manifest [hunk 4]
```diff
@@ -0,0 +1,3 @@
+rm -f manifest.xml
```
Comment (akrieger) [inline build-scripts/show-manifest:1]: If these are shell scripts can they end with .sh?

File: data/application_manifest.xml [hunk 5]
```diff
@@ -25,4 +20,9 @@
             </requestedPrivileges>
         </security>
     </trustInfo>
+    <asmv3:application>
+        <asmv3:windowsSettings xmlns="http://schemas.microsoft.com/SMI/2019/WindowsSettings">
+            <activeCodePage>UTF-8</activeCodePage>
+        </asmv3:windowsSettings>
+    </asmv3:application>
```
Comment (akrieger) [inline data/application_manifest.xml:27]: Why was this moved?

**Review by akrieger — COMMENTED** [review]
What do we need the raised minimum cmake version for?

**Comment by alef** [comment]
> What do we need the raised minimum cmake version for?

Because of this message:
```
CMake Deprecation Warning at src/third-party/CMakeLists.txt:1 (cmake_minimum_required):
  Compatibility with CMake < 3.5 will be removed from a future version of
  CMake.

  Update the VERSION argument <min> value or use a ...<max> suffix to tell
  CMake that the project does not need compatibility with older versions.
```

File: data/application_manifest.xml (hunk 5 above)
Comment (alef) [reply data/application_manifest.xml:27]: One of the linkers or resource compiler (don't remember which one on what platform) was reordering it in the final executable. So I moved the same way it had appear in that one.

File: build-scripts/show-manifest (hunk 3 above)
Comment (alef) [reply build-scripts/show-manifest:2]: This was a quick & dirty script for me to print the manifest, and not loosing it during a cleanup. I will remove from the repository.

**Comment by alef** [comment]
[92b1caa](https://github.com/CleverRaven/Cataclysm-DDA/pull/71089/commits/92b1caa86e3a87695c2825aa69d0eed55556fa44) not true. Still left out are:
- [ ] `libavif-16.dll`
- [ ] `libharfbuzz-0.dll`
- [ ] `libhwy.dll`
- [ ] `libogg-0.dll`

File: CMakeLists.txt [hunk 6]
```diff
@@ -94,6 +97,10 @@ endif ()
 
 include(CheckCXXCompilerFlag)
 
+if(CMAKE_HOST_SYSTEM_NAME STREQUAL "Windows" AND CMAKE_CXX_COMPILER_ID STREQUAL "GNU")
+    set(MSYS2 True)
```
Comment (Qrox) [inline CMakeLists.txt:101]: Could you rename this to something more generic? MSYS2 is, AFAIK, only one of several Windows GNU C toolchains. I for example use MinGW-w64 directly without using MSYS2 to compile the tiles build.

Comment (Qrox) [reply CMakeLists.txt:101]: I compile with CMake and set the compiler/linker executables through its command line. Some years ago I used `make` but I don't remember using `MSYS2=1` either.

Comment (alef) [reply CMakeLists.txt:101]: 44448f4 should do it.

**Comment by alef** [comment]
rebased
